town.logic

```diff
@@ -1,2 +1,116 @@
 area Town:
-  room Main
\ No newline at end of file
+  room Main:
+    node Anju:
+      chest AnjuHP
+    node AnjuGarden:
+      door Anju
+    node AnjuRoof:
+      door AnjuChimney
+    node CatsMinish:
+      door Fountain
+      door StratoMinishBack
+    node FigurineHouseDoorEast:
+      door FigurineHouseEast
+    node GormanHouseSouth:
+      door GormanHouseSouth
+    node HearthLedge:
+      door HearthLedge
+      chest HearthLedge
+    node JuliettaRoof:
+      door JuliettaChimney
+      door RomioChimneyEast
+    node LibraryEntrance:
+      entrance LibraryBottom
+    node LibraryExit:
+      exit LibraryBottom
+    node LibraryRoof:
+      door LibraryRoof
+    node LibraryRoofMinish:
+      door FlipperCave
+      door LibraryMinish
+    LibraryRoof -> LibraryRoofMinish: item Cane
+    node Main:
+      chest CafeLady
+      door Bakery
+      door Borlov
+      door East
+      door FigurineHouseWest
+      door HearthNorth
+      door HearthSouth
+      door GormanHouseWest
+      door Julietta
+      door LibraryBottom
+      door MayorWest
+      door North
+      door PostOffice
+      door Rem
+      door Romio
+      door SchoolWest
+      door Simon
+      door South
+      door Stockwell
+      door Strato
+      door Swiftblade
+      door Well
+      entrance Crest
+      flag Virtual.Zeffa.TownCrest
+    Main <-> Anju: (item Flippers | item Cape)
+    Main <-> FigurineHouseDoorEast: item CarlovMedal
+    Main <-> GormanHouseSouth: flag TalkedToOracle
+    Main -> LibraryEntrance: flag LibraryManHasMoved
+    LibraryExit-> Main
+    node MainBellHP:
+      chest MainBellHP
+    Main <-> MainBellHP: item Cape 
+    node MainMinish:
+      door AnjuMinish
+      door BakeryMinish
+      door CafeMinish
+      door CarpenterBuildingMinish
+      door DrLeftMinish
+      door HearthMinish
+      door RomeoMinish
+    node MayorGarden:
+      door MayorEast
+      door MayorBasementLadder
+    node MinishBridge:
+      door CafeBridgeMinish
+      door StockwellBridgeMinish
+    node RomioRoof:
+      door RomioChimneyWest
+    node SchoolGarden:
+      door SchoolEast
+    node SchoolGardenMinish:
+      door SchoolEntranceMinishEast
+      door SchoolGarden
+    node SchoolRoof:
+      chest SchoolRoof
+      door SchoolEntranceMinishWest
+    node SouthWest:
+      door Carlov
+      door Carpenters
+      door DrLeft
+    Main <-> SouthWest: (item Flippers | item Cape | flag DogMoves)
+    node TownDigging:
+      door TownDigging
+    Main <-> TownDigging: item MoleMitts
+    node WaterfallExit:
+      exit Waterfall
+    node WaterfallEntrance:
+      entrance Waterfall
+    Main -> WaterfallEntrance: (item Flippers & flag TownWaterfallFusion)
+    WaterfallExit -> Main: item Flippers
+    node West:
+      door West
+    Main <-> West: item CanSpin
```

Comment:
I know berryl said to check for spin here, but really it's another flag, that can be set here if you have spin, and makes the guard move aside. We'll probably also want the known skip for the guard (using boots) to be encoded - something like
```suggestion
    Main -> West: (flag GuardMoved | (setting TownGuardBootSkip & item Boots))
    West -> Main: (flag GuardMoved | setting TownGuardEastwards) # should moving past the guard this way be in logic? probably yes unless it's decoupled ER, this way we can choose that elsewhere
    node GuardMove:
      flag GuardMoved
    Main -> GuardMove: item CanSpin
```